### PR TITLE
Fix for PHP 7.3

### PR DESCRIPTION
**Pull request by remicollet**

### CODE REVIEW

File: zmq.c [hunk 1]
```diff
@@ -235,7 +235,11 @@ php_zmq_context *php_zmq_context_get(zend_long io_threads, zend_bool is_persiste
 		le.type = php_zmq_context_list_entry();
 		le.ptr  = context;
 
+#if PHP_VERSION_ID < 70300
 		GC_REFCOUNT(&le) = 1;
+#else
+		GC_SET_REFCOUNT(&le, 1);
+#endif
```
Comment (remicollet) [inline zmq.c]: From UPGRADING.INTERNALS
```
  f. GC_REFCOUNT() is turned into inline function and can't be modified direcly.
     All reference-counting operations should be done through corresponding
     macros GC_SET_REFCOUNT(), GC_ADDREF() and GC_DELREF().

     GC_REFCOUNT(p)++ should be changed into GC_ADDREF(p),
     GC_REFCOUNT(p)-- into GC_DELREF(p),
     GC_REFCOUNT(p) = 1 into GC_SET_REFCOUNT(p, 1).

```

File: zmq_helpers.c [hunk 2]
```diff
@@ -63,7 +63,7 @@ char *php_zmq_printable_func (zend_fcall_info *fci, zend_fcall_info_cache *fci_c
 	char *buffer = NULL;
 
 	if (fci->object) {
-		spprintf (&buffer, 0, "%s::%s", fci->object->ce->name->val, fci_cache->function_handler->common.function_name);
+		spprintf (&buffer, 0, "%s::%s", fci->object->ce->name->val, ZSTR_VAL(fci_cache->function_handler->common.function_name));
```
Comment (remicollet) [inline zmq_helpers.c]: this is a zend_string since 7.0.

PHP 7.3 now nicely raise -Wformat warning about this

File: zmq_pollset.c [hunk 3]
```diff
@@ -169,7 +169,7 @@ static
 zend_string *s_create_key(zval *entry)
 {
 	if (Z_TYPE_P(entry) == IS_RESOURCE) {
-		return strpprintf(0, "r:%ld", Z_RES_P(entry)->handle);
+		return strpprintf(0, "r:%d", Z_RES_P(entry)->handle);
```
Comment (remicollet) [inline zmq_pollset.c]: -Wformat again, hadle is an int

File: zmq_pollset.c [hunk 4]
```diff
@@ -222,7 +222,7 @@ size_t php_zmq_pollset_num_items(php_zmq_pollset *set)
 zend_string *php_zmq_pollset_add(php_zmq_pollset *set, zval *entry, int events, int *error) 
 {
 	zend_string *key;
-	size_t num_items, index;
+	size_t index;
```
Comment (remicollet) [inline zmq_pollset.c]: -Wunused-but-set-variable

**Comment by remicollet** [comment]
Notice: this only fix the build, there is something wrong, test suite segfault in various test...

**Comment by edhelas** [comment]
Is there some updates regarding this PR? The currently packaged version in Debian of PHP is 7.3 and it's not working anymore with the php-zmq package.

**Comment by edhelas** [comment]
I applied your patch and tried with Movim, still having a `zend_mm_heap corrupted` when using php-zmq.

**Comment by remicollet** [comment]
pr #193 is better

**Pull request closed without merging**